integration/feature/init/src/MillInitUtils.scala

```diff
@@ -0,0 +1,104 @@
+package mill.integration
+
+import mill.testkit.IntegrationTester
+import utest.*
+
+object MillInitUtils {
+  val defaultInitCommand =
+    Seq("init", "--base-module", "BaseModule", "--deps-object", "Deps", "--merge")
+
+  case class SplitResolvedTasks(
+      all: Seq[String],
+      successful: Seq[String],
+      failed: Seq[String]
+  ) {
+    val allSorted = all.sorted
+    {
+      val successfulAndFailedSorted = (successful ++ failed).sorted
+      require(allSorted == successfulAndFailedSorted, s"$allSorted != $successfulAndFailedSorted")
+    }
+  }
+  object SplitResolvedTasks {
+    def apply(
+        all: Seq[String] | Null = null,
+        successful: Seq[String] | Null = null,
+        failed: Seq[String] | Null = null
+    ) =
+      new SplitResolvedTasks(
+        if (all != null) all else successful ++ failed,
+        if (successful != null) successful else all diff failed,
+        if (failed != null) failed else all diff successful
+      )
+  }
+
+  /**
+   * @param expectedCompileTasks [[ None ]] to denote that the `resolve __.compile` task fails
+   * @param expectedTestTasks [[ None ]] to denote that the `resolve __.test` task fails
+   * @return
+   */
+  def testMillInit(
+      tester: IntegrationTester,
+      initCommand: Seq[String] = defaultInitCommand,
+      modifyConvertedBuild: () => Unit = () => (),
+      expectedInitResult: Boolean = true,
+      // expectedCompileResult: Boolean,
+      expectedCompileTasks: Option[SplitResolvedTasks],
+      expectedTestTasks: Option[SplitResolvedTasks]
+  ) = {
+    import tester.*
+
+    val initResult = eval(initCommand, stdout = os.Inherit, stderr = os.Inherit)
```

Comment:
Could we not stream the output via `os.Inherit` by default, and only print it on failure? It's very verbose and causing Github Actions to have problems in my browser